### PR TITLE
Fix webapp configs

### PR DESCRIPTION
**Pull request by jimallman**
Cleanup and consolidation of Ansible configuration of our web-apps

- simplify API base URLs and endpoints
- add explicit production API base URLs (used in dev deployment for some API calls)
- rename some confusing Ansible variables (and filenames) and web-app config keys
- check current playbook's roles (via `role_names`) to adjust apache config for API vs. web-app vs. others TBD

NB - This is working on `devtree`, along with [a companion`fix-webapp-configs` PR in opentree](https://github.com/OpenTreeOfLife/opentree/pull/1312).

### CODE REVIEW

File: roles/apache/tasks/main.yml [hunk 1]
```diff
@@ -52,10 +52,9 @@
     - restart apache2
 
 
-## RIGHT NOW THIS WILL ONLY WORK FOR API!
 - name: write /etc/apache2/opentree-shared.conf
   template:
-    src: templates/opentree_devapi_conf.j2
+    src: "{{ 'templates/opentree_devapi_conf.j2' if 'phylesystemapi' in role_names else 'templates/opentree_shared_conf.j2' }}"
```
Comment (jimallman) [inline roles/apache/tasks/main.yml]: @snacktavish This test of `... in role_names` seems to work well (see also debug output in subsequent tasks).

**Comment by jimallman** [comment]
This could probably use a final review for our always-on-production endpoints. If necessary, we can use both dev (default) and production versions of a method, by defining two different endpoints [in our config](https://github.com/OpenTreeOfLife/ot-ansible/pull/32/files#diff-bb366cb4fc45fe2453381a0f70634d53947c2f7b4e61b8f3f79f980ade98d5f6), like so (and using both names as appropriate in the web-app code):
```inputrc
# a typical endpoint, on both default/dev and production
getContextForNames_url = {default_apis}/v3/tnrs/infer_context
PRODUCTION_getContextForNames_url = {production_apis}/v3/tnrs/infer_context

# and this is a cached call for both default/dev and production
getSynthesisSourceList_url = {CACHED_default_apis}/v3/tree_of_life/about
PRODUCTION_getSynthesisSourceList_url = {CACHED_production_apis}/v3/tree_of_life/about
```

**Comment by snacktavish** [comment]
This branch was merged into development - can we delete this branch "fix-webapp-configs" and work from development?

**Comment by snacktavish** [comment]
I merged most of these into development, and used that to deploy production phylesystem-api!
Can we move the webhook deploy commit to a new branch off of development?

**Comment by jimallman** [comment]
> Can we move the webhook deploy commit to a new branch off of development?

Agreed. Moving these changes to a new branch, and closing this one.

**Pull request closed without merging**